tests/unit/SpecHelper.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+const iframes = [];
+
+/**
+ * Create a doc for unit tests.
+ * @param {string[]} plugins Paths of plugins to load and run. Example: `/src/core/algorithms.js`
+ * @param {object} config JSON-serializable respecConfig object.
+ * @return {Promise<Document>}
+ */
+export function makePluginDoc(
+  plugins,
+  config = {},
+  { head = `<meta charset="UTF-8" />`, body = "" } = {}
+) {
+  const html = `
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        ${head}
+        <script>
+          var respecConfig = ${JSON.stringify(config || {}, null, 2)};
+        </script>
+        <script type="module">
+          const plugins = ${JSON.stringify(plugins)};
```

Comment:
Might be better to wrap this in a self-invoking  `async` and the sprinkle some `await`s around... it's hard to follow the logic around error handling as it currently is - and try/catch will be easier on the eyes.